blackbox/docs/sql/optimize.txt

```diff
@@ -0,0 +1,95 @@
+.. highlight:: psql
+.. _optimize:
+
+========
+Optimize
+========
+
+In Crate every table (or if partitioned every partition) consists of segments.
+When inserting/deleting/updating data new segments are created and there is the
+danger of having the number of segments grow too big. Each segment consumes file
+handles, memory and CPU and on top of that the performance of a search request
+decreases as the number of segments visited increase. Crate solves this problem
+by merging segments automatically in the background. Small segments are merged
+into bigger segments, which, in turn, are merged into even bigger segments.
+Moreover any deleted rows and documents are not copied to the new bigger segment
+during this process.
+
+.. note::
+
+System tables and blob tables cannot be optimized
```

Comment:
I'd move this note further down after the example optimize example. It interrupts the reading flow otherwise